File: frontend/src/types/priceUpdate.ts
export type PriceUpdateStatus = 'calculated' | 'applied' | 'zeroed' | 'will_zero' | 'error';

export type TariffItem = { type: string; amount: number; pct?: number | null; [key: string]: unknown };

export type PriceUpdate = {
  id: number;
  store_id: number;
  sku: string;
  supplier: string | null;
  supplier_price: number | null;
  old_price: number | null;
  new_price: number;
  difference: number | null;
  difference_pct: number | null;
  profit: number | null;
  actual_roi: number | null;
  tariffs_json: string | null;
  old_stock: number | null;
  new_stock: number | null;
  requires_confirmation: boolean;
  status: PriceUpdateStatus;
  created_at: string | null;
};

File: backend/models/price_update.py
from sqlalchemy import Boolean, Column, DateTime, Float, Integer, String, Text, func

from database.db import Base


class PriceUpdate(Base):
    __tablename__ = "price_updates"

    id = Column(Integer, primary_key=True, index=True)
    store_id = Column(Integer, nullable=False, index=True)
    sku = Column(String, nullable=False)
    supplier = Column(String)                      # имя поставщика
    supplier_price = Column(Float)                 # закупочная цена
    old_price = Column(Float)
    new_price = Column(Float, nullable=False)
    difference = Column(Float)
    difference_pct = Column(Float)                 # изменение в %
    profit = Column(Float)                         # прибыль в рублях
    actual_roi = Column(Float)                     # фактический ROI в %
    tariffs_json = Column(Text)                    # JSON-список тарифов ЯМ
    old_stock = Column(Integer)                    # остаток до применения
    new_stock = Column(Integer)                    # остаток после применения
    requires_confirmation = Column(Boolean, default=False)
    status = Column(String, default="calculated")  # calculated | applied | zeroed | error
    created_at = Column(DateTime(timezone=True), server_default=func.now())
